### PR TITLE
feat: libarchive 3.7.5.bcr1 for windows compression filter fixes

### PR DESCRIPTION
**Pull request by thesayyn**
Fixes linking of various filters for windows making it on par with other platforms. 

Currently, when compiling libarchive for windows, some define statements are missing from windows_config.h which results in libarchive shelling out to programs such as `gzip` `xz` when encountering archives with such compression algos. 

Running `bazel-bin/external/libarchive~/tar/bsdtar.exe -tf test.tar.gz`

eg: version 3.7.5

```PS C:\Users\Sahin\Documents\bsdtar-prebuilt>  bazel-bin/external/libarchive~/tar/bsdtar.exe -tf test.tar.gz
bsdtar.exe: Error opening archive: Can't initialize filter; unable to run program "gzip -d"
```

version 3.7.5.bcr.1 (this patch):

```
.github/
.github/workflows/
.github/workflows/build.yaml
.github/workflows/release.yml
```

### CODE REVIEW

**Comment by thesayyn** [comment]
https://github.com/aspect-build/rules_js/issues/1739#issuecomment-2455809190

**Comment by zaucy** [comment]
Hm thats fair. Thanks for fixing and investigating the issue. The defines I generated were taken from a naive cmake build that certainly wouldn't have covered every environment on Windows. I noticed in the patch that the fuzz test had the 'manual' tag added. It's already skipped in the presubmit, it shouldn't need that manual tag.

`test_read_truncated_filter` skip presubmit tag removed - last time I ran it on the presubmit it took a very long time. I suppose when you fix the validation error we'll see how long it is!

`test_write_filter_gzip_timestamp` skip presubmit tag removed - glad to see that this test works now :)

As as side note - we should probably move to an overlay style patch for libarchive sooner than later since the patch is quite large and mostly additional. Would have probably made your adjustments much easier and easier for reviewing the changes.

**Comment by thesayyn** [comment]
> As as side note - we should probably move to an overlay style patch for libarchive sooner than later since the patch is quite large and mostly additional. Would have probably made your adjustments much easier and easier for reviewing the changes.

Definitely agree. We need a better pattern here. 

> It's already skipped in the presubmit, it shouldn't need that manual tag.

I added it to skip it on local as well, happy to remove it if it needs to be removed.

**Comment by thesayyn** [comment]
This should be ready now, i updated the patch to exclude MODULE.bazel.lock and test.patch, :sigh:

**Comment by thesayyn** [comment]
> we should probably move to an overlay style patch for libarchive sooner than later since the patch is quite large and mostly additional.

okay after lots of pain and suffering, i have overlay thing setup and good to go. PTAL @zaucy @dzbarsky

File: modules/libarchive/3.7.5.bcr.1/overlay/libarchive/test/BUILD.bazel
Comment (dzbarsky) [inline modules/libarchive/3.7.5.bcr.1/overlay/libarchive/test/BUILD.bazel:446]: Maybe just tag it incompatible on Windows so it runs on others?

File: modules/libarchive/3.7.5.bcr.1/overlay/libarchive/test/BUILD.bazel
Comment (dzbarsky) [inline modules/libarchive/3.7.5.bcr.1/overlay/libarchive/test/BUILD.bazel:797]: Ditto on these (compatibility)

**Review by dzbarsky — APPROVED** [review]
Nice, so much easier to read :)

**Review by bazel-io — DISMISSED** [review]
Hello @bazelbuild/bcr-maintainers, all modules in this PR have been approved by their maintainers. Please take a final look to merge this PR.

File: modules/libarchive/3.7.5.bcr.1/overlay/libarchive_bazel_windows_config.h
Comment (zaucy) [inline modules/libarchive/3.7.5.bcr.1/overlay/libarchive_bazel_windows_config.h:117]: looks like it was missed on other patches, but these version defines should probably be updated 😅

File: modules/libarchive/3.7.5.bcr.1/overlay/libarchive/test/BUILD.bazel
Comment (thesayyn) [reply modules/libarchive/3.7.5.bcr.1/overlay/libarchive/test/BUILD.bazel:446]: Hmm, this wasn't me, but happy do that.

File: modules/libarchive/3.7.5.bcr.1/MODULE.bazel
Comment (thesayyn) [reply modules/libarchive/3.7.5.bcr.1/MODULE.bazel:4]: done

File: modules/libarchive/3.7.5.bcr.1/overlay/libarchive_bazel_windows_config.h
Comment (thesayyn) [reply modules/libarchive/3.7.5.bcr.1/overlay/libarchive_bazel_windows_config.h]: done

File: modules/libarchive/3.7.5.bcr.1/overlay/libarchive/test/BUILD.bazel
Comment (zaucy) [reply modules/libarchive/3.7.5.bcr.1/overlay/libarchive/test/BUILD.bazel:446]: that was me initially - these are only test targets and are not public facing so I wouldn't be too worried about it. If you'd like to do it then great, but I wouldn't require the change myself.

Comment (thesayyn) [reply modules/libarchive/3.7.5.bcr.1/overlay/libarchive/test/BUILD.bazel:446]: or can i leave this to somebody else as i need to fix zstd as well.

File: modules/libarchive/3.7.5.bcr.1/presubmit.yml
Comment (zaucy) [inline modules/libarchive/3.7.5.bcr.1/presubmit.yml:12]: i think the presubmit is failing due to the bazel compability change. Removing `6.x` and adjusting the 7 to be `>=7.2.1` (assuming the presubmit syntax is the same)

Comment (thesayyn) [reply modules/libarchive/3.7.5.bcr.1/presubmit.yml]: done

**Review by dzbarsky — APPROVED** [review]

**Review by bazel-io — APPROVED** [review]
Hello @bazelbuild/bcr-maintainers, all modules in this PR have been approved by their maintainers. Please take a final look to merge this PR.

**Review by zaucy — APPROVED** [review]

**Pull request merged**
